tests/fixtures/controllers/responseHeaderController.ts

```diff
@@ -0,0 +1,53 @@
+import { Controller, Get, Route, SuccessResponse, Response, TsoaResponse, Res } from '@tsoa/runtime';
+
+/**
+ * asdasdsda
+ */
+class ResponseHeader {
+  /**
+   * a link string
+   */
+  Link: string;
+
+  /**
+   * b link str
+   */
+  LinkB: string;
+}
+
+@Route('ResponseHeader')
+export class ResponseHeaderController extends Controller {
+  @Get('SuccessResponseWithHeaderClass')
+  @SuccessResponse<ResponseHeader>(200, 'zzz')
+  public async handler(): Promise<void> {
+    return;
+  }
+
+  @Get('SuccessResponseWithObject')
+  @SuccessResponse<{ linkA: string; linkB: string }>(200, 'xxx')
+  public async handler2(): Promise<void> {
+    return;
+  }
+
+  @Get('ResponseWithHeaderClass')
+  @Response<null, ResponseHeader>(200, 'yyy')
+  public async handler3(): Promise<void> {
+    return;
+  }
+
+  @Get('ResponseWithObject')
+  @Response<null, { linkC: number; linkD: string }>(200, 'yyy')
+  public async handler4(): Promise<void> {
+    return;
+  }
+
+  @Get('TsoaResponseWithHeaderClass')
+  public async handler5(@Res() res: TsoaResponse<200, null, ResponseHeader>): Promise<void> {
+    return;
+  }
+
+  @Get('TsoaResponseWithObject')
+  public async handler6(@Res() res: TsoaResponse<200, null, { linkE: string; linkF: number }>): Promise<void> {
+    return;
+  }
```

Comment:
```ts
interface ComplexHeader {
  /**
   * Numbers description
   * @example [1,2,3]
   * @default [1,2,3]
   * @isInt
   */
  nums: number[];
}


  @Get('TsoaResponseWithHeaderArray')
  public async handler7(@Res() res: TsoaResponse<200, string, ComplexHeader[]>) {
    return res(200, 'Hello, World', [{ nums: [1, 2, 3] }]);
  }
```

So we can test that all the possible header object fields (using the style + schema, which is what req headers do): 
- description
- required
- ~~deprecated~~ (does not exist on property yet, so omit that for now)
- schema
- style (this should be static for all headers)
- allowReserved this should be static for all headers)
- ~~example(s)~~ (should be in the schema's example prop)

Reply:
@WoH You mean we can make multiple different header class(or field inside header class) to test all of these features with proper test cases?

Reply:
As long as every one of these properties is tested once, we are good imo.
The only remaining properties are `style` and `allowReserved`, which we should set to provide the best spec when it comes to returning header properties of `string[]`.